daemon/daemon.go

```diff
@@ -405,19 +405,19 @@ func (d *Daemon) init() error {
 	fw.WriteString(common.FmtDefineArray("ROUTER_IP", d.conf.NodeAddress.IPv6Address))
 
 	ipv4GW := d.conf.NodeAddress.IPv4Address
-	fmt.Fprintf(fw, "#define IPV4_GATEWAY %#x\n", binary.LittleEndian.Uint32(ipv4GW))
+	fmt.Fprintf(fw, "#define IPV4_GATEWAY %#x\n", byteorder.NetworkToHostU32(ipv4GW))
```

Comment:
Should this one line be fixed separately first, if it hasn't already in master(need to check)?

Currently in the process of slicing the patch up into a series where more tests are added.
Will respin with feedback addressed later today.